core/trino-main/src/main/java/io/trino/sql/planner/optimizations/AddExchanges.java

```diff
@@ -643,7 +643,7 @@ public PlanWithProperties visitLimit(LimitNode node, PreferredProperties preferr
 
             PlanWithProperties child = planChild(node, PreferredProperties.any());
 
-            if (!child.getProperties().isSingleNode()) {
+            if (!node.isPartial() && !child.getProperties().isSingleNode()) {
```

Comment:
Why it's not beneficial to pushdown limit below exchange even for partial? Seems it doesn't hurt but could reduce network traffic.

Reply:
If the current node is already a partial limit, then we probably don't want to insert a full gathering exchange for each of those since:
1. The current node is already a partial limit and will still be present in the plan, so we still have partial limit enforcement
2. The parent final limit will insert a gathering exchange anyway, so no need to do a separate remote gathering exchange for each partial limit child (this will actually increase the amount of exchanged data in most plan shapes I can think of).